assets/src/edit-story/components/reorderable/useScroll.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useCallback, useRef, useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import useBatchingCallback from '../../utils/useBatchingCallback';
+
+const SCROLL_DISTANCE = 5;
+
+function useScroll(isReordering) {
+  const [hasScrollAbove, setHasScrollAbove] = useState(false);
+  const [hasScrollBelow, setHasScrollBelow] = useState(false);
+  const [scrollDirection, setScrollDirection] = useState(0);
+  const scrollTarget = useRef(null);
+
+  const setScrollTarget = useCallback(
+    (node) => (scrollTarget.current = node.parentElement),
```

Comment:
This currently breaks the document tab on the right hand side ⚠️ 

```
Uncaught TypeError: Cannot read property 'parentElement' of null
```

Reply:
Just saw the same thing trying to work on the Document Panel.

Looks like a test is missing for the tabs to ensure these at least open without erroring.